new-log-viewer/src/services/LogFileManager.ts

```diff
@@ -0,0 +1,200 @@
+import {
+    Decoder,
+    DecoderOptionsType,
+} from "../typings/decoders";
+import {MAX_V8_STRING_LENGTH} from "../typings/js";
+import {
+    BeginLineNumToLogEventNumMap,
+    CURSOR_CODE,
+    CursorType,
+    FileSrcType,
+} from "../typings/worker";
+import {getUint8ArrayFrom} from "../utils/http";
+import {formatSizeInBytes} from "../utils/units";
+import {getBasenameFromUrlOrDefault} from "../utils/url";
+import JsonlDecoder from "./decoders/JsonlDecoder";
+
+
+/**
+ * Class to manage the retrieval and decoding of a given log file.
+ */
+class LogFileManager {
+    numEvents: number = 0;
+
+    readonly #pageSize: number;
+
+    #fileData: Uint8Array | null = null;
+
+    #fileName: string | null = null;
+
+    #decoder: Decoder | null = null;
+
+    constructor (pageSize: number) {
+        this.#pageSize = pageSize;
+    }
+
+    /**
+     * Creates a new LogFileManager.
+     *
+     * @param fileSrc The source of the file to load. This can be a string representing a URL, or a
+     * File object.
+     * @param pageSize Page size for setting up pagination.
+     * @param decoderOptions Initial decoder options.
+     * @return A Promise that resolves to the created LogFileManager instance.
+     */
+    static async create (
+        fileSrc: FileSrcType,
+        pageSize: number,
+        decoderOptions: DecoderOptionsType
+    ): Promise<LogFileManager> {
+        const mgr = new LogFileManager(pageSize);
+        await mgr.#loadFile(fileSrc);
```

Comment:
As discussed offline, let's make this a static/non-class method, collect the results here, and then pass them into the `LogFileManager` constructor. That would also allow us to get rid of the "is #loadFile" called checks in other methods.